deployer/deployer-services/src/main/java/io/hyscale/deployer/services/deployer/impl/KubernetesDeployer.java

```diff
@@ -169,6 +192,36 @@ public boolean authenticate(AuthConfig authConfig) {
         // TODO
         return false;
     }
+    
+    
+	@Override
+	public void exec(K8sAuthorisation k8sAuthorisation, String servicename, String appname, String namespace,
+			String replicaName) {
+
+		List<String> commands = null;
+		KubeConfig kubeConfig = null;
+		if (k8sAuthorisation != null && k8sAuthorisation.getK8sAuthType() == K8sAuthType.BASIC_AUTH) {
+			K8sBasicAuth k8sBasicAuth = (K8sBasicAuth) k8sAuthorisation;
+			commands = ExecCommandUtil.getExecCommand(replicaName, namespace, k8sBasicAuth);
+		} else if (k8sAuthorisation != null && k8sAuthorisation.getK8sAuthType() == K8sAuthType.KUBE_CONFIG_FILE) {
+			K8sConfigFileAuth k8sConfigFileAuth = (K8sConfigFileAuth) k8sAuthorisation;
+			kubeConfig = getKubeConfig(k8sConfigFileAuth);
+			if (kubeConfig.getAccessToken() != null) {
+				commands = ExecCommandUtil.getExecCommand(replicaName, namespace, kubeConfig.getAccessToken());
+				CommandExecutor.executeExec(commands);
+			}
+		}
+	}
+
+	private KubeConfig getKubeConfig(K8sConfigFileAuth k8sConfigFileAuth) {
```

Comment:
As discussed this should not be required right? Directly passing the file to the command works?